MQTT: Use subscription identifiers. Fixes #8879.

wilx opened the pull request.

wilx committed:
MQTT: Use subscription identifiers. Fixes #8879.

pivotal-cla commented:
@wilx Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-integration&pullRequestId=8880)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-integration&pullRequestId=8880) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

artembilan commented:
Would you mind to issue this PR against `main` instead, please?

We still need to apply the change over there as well.
And then we will back-port it to all the supported versions.

Thanks

wilx committed:
MQTT: Use subscription identifiers.

Add a test to check that Mqttv5PahoMessageDrivenChannelAdapter
actually receives messages via a shared topic.

artembilan reviewed (changes requested):
Thank you for the test!
However I still would like to see the PR agains `main`.
